bundles/CoreBundle/Migrations/Version20230814142319.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20230814142319 extends AbstractMigration
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function getDescription(): string
+    {
+        return 'Add `dataModificationDate` column to `assets` database table';
+    }
+
+    public function up(Schema $schema): void
+    {
+        if ($schema->getTable('assets')->hasColumn('dataModificationDate') === false) {
+            $this->addSql('ALTER TABLE `assets`
+                ADD COLUMN `dataModificationDate` INT(11) UNSIGNED DEFAULT NULL AFTER `modificationDate`;'
+            );
```

Comment:
Copy the current `modificationDate` so the values get populated? Alternatively, a command to populate it from the filesystem.

Reply:
we don't want to have the duplicate info from `modificationDate` that increases the table size unnecessarily since the null check works at the moment. 

Also, adding a command is not useful since this needs to be run just once. If you like, you can provide a Gist of the script to populate the field from filesystem 😊

Reply:
Functionally it's identical since the bug triggering will fill in the timestamp, but OK, I guess that's good enough too, this was meant as an optimization to avoid triggering it one more time after this bugfix getting deployed.